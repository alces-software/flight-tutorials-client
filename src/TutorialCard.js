/*=============================================================================
 * Copyright (C) 2017 Stephen F. Norledge and Alces Flight Ltd.
 *
 * This file is part of Flight Launch.
 *
 * All rights reserved, see LICENSE.txt.
 *===========================================================================*/
import React from 'react';

import Card from './Card';
import Markdown from 'react-markdown';
import Overlay from './TutorialCardOverlay';
import './styles/TutorialCard.scss';

const TutorialCard = ({
  onSelectTutorial,
  tutorial,
}) => (
  <Card
    className="TutorialCard"
    onClick={onSelectTutorial}
    subtitle={tutorial.subtitle}
    subtitleSize="medium"
    title={tutorial.title}
    titleLogoOnRight
    titleLogoUrl={tutorial.logoUrl}
    titleSize="large"
  >
    <Markdown source={tutorial.description} />
    <Overlay onSelectTutorial={onSelectTutorial} />
  </Card>
);

export default TutorialCard;
