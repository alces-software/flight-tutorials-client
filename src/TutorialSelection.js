// @flow
/*=============================================================================
 * Copyright (C) 2017 Stephen F. Norledge and Alces Flight Ltd.
 *
 * This file is part of Flight Compute Tutorials.
 *
 * All rights reserved, see LICENSE.txt.
 *===========================================================================*/
import React from 'react';

import type { TutorialType } from './types';

const TutorialSelection = ({
  onSelectTutorial,
  tutorials,
}: {
  onSelectTutorial: (number) => void,
  tutorials: Array<TutorialType>
}) => {
  const options = tutorials.map((t, idx) => <TutorialOption
    key={idx}
    tutorial={t}
    onSelectTutorial={() => onSelectTutorial(idx)}
  />);

  return (
    <div>
      <p>Which tutorial do you wish to undertake?</p>
      {options}
    </div>
  );
};

const TutorialOption = ({
  onSelectTutorial,
  tutorial,
}:{
  onSelectTutorial: () => void,
  tutorial: TutorialType
}) => {
  return (
    <div>
      <h3>{tutorial.title}</h3>
      <div dangerouslySetInnerHTML={{__html: tutorial.description}} />
      <button onClick={onSelectTutorial}>
        Select
      </button>
    </div>
  );
};

export default TutorialSelection;
