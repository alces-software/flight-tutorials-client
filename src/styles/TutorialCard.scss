/*=============================================================================
 * Copyright (C) 2016 Stephen F. Norledge and Alces Flight Ltd.
 *
 * This file is part of Flight Tutorials.
 *
 * All rights reserved, see LICENSE.txt.
 *===========================================================================*/

$card-width: 560px;
$card-height: 425px;

.TutorialCard {
    width: $card-width;
    height: $card-height;
}
