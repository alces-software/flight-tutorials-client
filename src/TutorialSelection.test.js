// @flow
/*=============================================================================
 * Copyright (C) 2017 Stephen F. Norledge and Alces Flight Ltd.
 *
 * This file is part of Flight Tutorials.
 *
 * All rights reserved, see LICENSE.txt.
 *===========================================================================*/
import React from 'react';
import { render } from 'react-dom';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import TutorialSelection from './TutorialSelection';
import TutorialCard from './TutorialCard';

const tutorial1 = {
  title: 'Tutorial 1',
  description: 'Tutorial 1 description',
  firstStep: 'step1',
  steps: {
    step1: {
      title: 'Tutorial 1 step 1',
      description: 'Tutorial 1 step 1 description',
      matches: [],
    },
    step2: {
      title: 'Tutorial 1 step 2',
      description: 'Tutorial 1 step 2 description',
      matches: [],
    },
  },
};

const tutorial2 = {
  title: 'Tutorial 2',
  description: 'Tutorial 2 description',
  firstStep: 'step1',
  steps: {
    step1: {
      title: 'Tutorial 2 step 1',
      description: 'Tutorial 2 step 1 description',
      matches: [],
    },
    step2: {
      title: 'Tutorial 2 step 2',
      description: 'Tutorial 2 step 2 description',
      matches: [],
    },
  },
};

const tutorials = [tutorial1, tutorial2];
const onSelectTutorialSpy = jest.fn();

const renderTutorialSelection = () => (
  <TutorialSelection
    onSelectTutorial={onSelectTutorialSpy}
    tutorials={tutorials}
  />
);

it('renders without crashing', () => {
  const node = document.createElement('div');
  render(renderTutorialSelection(), node);
});

it('renders correctly', () => {
  const tree = renderer.create(
    renderTutorialSelection(),
  ).toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders each tutorial once', () => {
  const wrapper = shallow(renderTutorialSelection());

  expect(wrapper.find(TutorialCard).length).toEqual(2);
  expect(wrapper.find(TutorialCard).at(0)).toHaveProp('tutorial', tutorial1);
  expect(wrapper.find(TutorialCard).at(1)).toHaveProp('tutorial', tutorial2);
});

it('calls onSelectTutorial when tutorial is selected', () => {
  const wrapper = shallow(renderTutorialSelection());

  tutorials.forEach((tutorial, idx) => {
    const option = wrapper.find(TutorialCard).at(idx);
    const button = option.dive().find('button').first();

    button.simulate('click');

    expect(onSelectTutorialSpy).toHaveBeenCalledWith(idx);
  });
});
