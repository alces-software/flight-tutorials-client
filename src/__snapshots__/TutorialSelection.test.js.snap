// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <p>
    Which tutorial do you wish to undertake?
  </p>
  <div>
    <h3>
      Tutorial 1
    </h3>
    <div>
      <p>
        Tutorial 1 description
      </p>
    </div>
    <button
      onClick={[Function]}
    >
      Select
    </button>
  </div>
  <div>
    <h3>
      Tutorial 2
    </h3>
    <div>
      <p>
        Tutorial 2 description
      </p>
    </div>
    <button
      onClick={[Function]}
    >
      Select
    </button>
  </div>
</div>
`;
